compiler/rustc_middle/src/ty/context/tls.rs

```diff
@@ -191,7 +193,11 @@ where
     F: FnOnce(&ImplicitCtxt<'_, 'tcx>) -> R,
 {
     with_context(|context| unsafe {
-        assert!(ptr_eq(context.tcx.gcx, tcx.gcx));
+        assert!(ptr::eq(
```

Comment:
Maybe add back the comment that casting to a pointer is because these are type-erased?